CFC_DataCollector/recommender/common.py

```diff
@@ -1,75 +1,108 @@
 from __future__ import division
 import urllib2
-import xml.etree.ElementTree as ET
 from get_database import *
 from datetime import datetime, timedelta
-import random
 import json
-import jsonpickle
-from trip import E_Mission_Trip
+from trip import *
+import copy
+import uuid
+import xml.etree.ElementTree as ET
+
+DATE_FORMAT = "%Y%m%dT%H%M%S-%W00" #This is a great hack thought of by Shaun
+CLASS_UUIDS = {
+  'b0d937d0-70ef-305e-9563-440369012b39': "Shankari's Husband",
+  '0763de67-f61e-3f5d-90e7-518e69793954': "Shankari Android",
+  '3a307244-ecf1-3e6e-a9a7-3aaf101b40fa': "Shankari iPhone",
+  '6245318c-d337-3530-9001-6b175dab73a7': "Jeff",
+  '1a65368b-888e-3e77-8f7c-1128f16da1df': "Gautham",
+  '5ecc845a-dbca-376f-8fb1-577bd7b18859': "Zack",
+  '6433c8cf-c4c5-3741-9144-5905379ece6e': "Jimmy",
+  'cc7f2ff0-8e73-3cfa-ab4c-647ebf025e42': "Shaun",
+  'f8fee20c-0f32-359d-ba75-bce97a7ac83b': "Shanthi"
+}
 
-DATE_FORMAT = "%Y%m%dT%H%M%S-0700"
+# Helper for development
+def uuid_to_name(uuid):
+  return CLASS_UUIDS.get(uuid, "Unknown UUID")
 
 def get_uuid_list():
-	uuid_list = [ ]
-	db = get_section_db()
-	for x in db.find():
-   		uuid_list.append(x['_id'])
-	return uuid_list
-'''
-def insert_into_pdb(pdb, my_id, new_perturbed_trip):
-	my_id = str(my_id)
-	assert(type(my_id) == str)
-	new_id = my_id.replace('.', "")
-	assert(type(new_id) == str)
-	if '.' not in new_id:
-		pdb.insert({new_id : new_perturbed_trip})
-'''
-    
-def initialize_empty_perturbed_trips(_id, pdb):
-	db = get_trip_db()
-	json_trip = db.find_one({"_id" : _id})
-	new_perturbed_trip = { }
-        if json_trip:
-	    trip = E_Mission_Trip.trip_from_json(json_trip)
-	    for pert in find_perturbed_trips(trip):
-	    	pert._id = pert._id.replace('.', '') 
-	    	new_perturbed_trip[pert._id] = None
-	    #insert_into_pdb(pdb, trip._id, new_perturbed_trip)
-	    _id = _id.replace('.', "")
-	    to_insert = { }
-	    to_insert['_id'] = _id
-	    to_insert['trips'] = new_perturbed_trip
-	    pdb.insert({'our_id': _id, "trips" : new_perturbed_trip})
+    '''
+    uuids = set()
+    #uuids.add(uuid.UUID('cc7f2ff0-8e73-3cfa-ab4c-647ebf025e42'))
+    uuids.add(uuid.UUID('6433c8cf-c4c5-3741-9144-5905379ece6e'))
+    '''
+    uuids = set()
+    db = get_trip_db()
+    for x in db.find():
+        uuids.add(x['user_id'])
+    return uuids
 
+def get_training_uuid_list():
+    '''
+    uuids = set()
+    #uuids.add(uuid.UUID('cc7f2ff0-8e73-3cfa-ab4c-647ebf025e42'))
+    uuids.add(uuid.UUID('6433c8cf-c4c5-3741-9144-5905379ece6e'))
+    '''
+    uuids = set()
+    db = get_trip_db()
+    for x in db.find():
+        uuids.add(x['user_id'])
+    return uuids
 
-def update_perturbations(_id, perturbed_trip):
-	db = get_perturbed_trips_db()
-	_id = _id.replace('.', '')
-	json_trip = db.find_one({"our_id" : _id})
-	json_trip[perturbed_trip._id] = jsonpickle.encode(perturbed_trip)
+def get_recommender_uuid_list():
+    uuids = set()
```

Comment:
@jimmydief @sdsingh, this is extremely slow. The correct way is to use "distinct" instead.